File: app/src/main/java/com/gooseco/technicalrulesquiz/data/model/QuizQuestion.kt
package com.gooseco.technicalrulesquiz.data.model

import com.google.gson.annotations.SerializedName

data class QuizQuestion(
    @SerializedName("id")
    val id: Int,

    @SerializedName("question")
    val question: String,

    @SerializedName("options")
    val options: List<String>,

    @SerializedName("correctAnswerIndex")
    val correctAnswerIndex: Int,

    @SerializedName("ruleReference")
    val ruleReference: RuleReference,

    @SerializedName("explanation")
    val explanation: String? = null,

    @SerializedName("ruleQuote")
    val ruleQuote: String? = null,

    @SerializedName("status")
    val status: String = "pending"  // "pending", "validated", "review_later", "rejected"
)

data class RuleReference(
    @SerializedName("section")
    val section: String,

    @SerializedName("subsection")
    val subsection: String? = null,

    @SerializedName("ruleNumber")
    val ruleNumber: String,

    @SerializedName("pageNumber")
    val pageNumber: Int? = null
) {
    fun getFullReference(): String {
        val parts = mutableListOf<String>()
        parts.add("Section $section")
        subsection?.let { parts.add(it) }
        parts.add("Rule $ruleNumber")
        pageNumber?.let { parts.add("(Page $it)") }
        return parts.joinToString(" - ")
    }
}

File: ios/PowerliftingQuiz/Sources/Models/QuizQuestion.swift
import Foundation

struct QuizQuestion: Codable, Identifiable {
    let id: Int
    let question: String
    let options: [String]
    let correctAnswerIndex: Int
    let ruleReference: RuleReference
    let explanation: String?
    let ruleQuote: String?
    let status: String

    init(from decoder: any Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        id = try container.decode(Int.self, forKey: .id)
        question = try container.decode(String.self, forKey: .question)
        options = try container.decode([String].self, forKey: .options)
        correctAnswerIndex = try container.decode(Int.self, forKey: .correctAnswerIndex)
        ruleReference = try container.decode(RuleReference.self, forKey: .ruleReference)
        explanation = try container.decodeIfPresent(String.self, forKey: .explanation)
        ruleQuote = try container.decodeIfPresent(String.self, forKey: .ruleQuote)
        status = (try container.decodeIfPresent(String.self, forKey: .status)) ?? "pending"
    }
}

struct RuleReference: Codable {
    let section: String
    let subsection: String?
    let ruleNumber: String
    let pageNumber: Int?

    var fullReference: String {
        var parts = ["Section \(section)"]
        if let sub = subsection { parts.append(sub) }
        parts.append("Rule \(ruleNumber)")
        if let page = pageNumber { parts.append("(Page \(page))") }
        return parts.joined(separator: " - ")
    }
}
